fdbserver/include/fdbserver/DDTxnProcessor.h

```diff
@@ -129,6 +144,20 @@ class DDTxnProcessor : public IDDTxnProcessor {
 	}
 
 	Future<Void> moveKeys(const MoveKeysParams& params) const override { return ::moveKeys(cx, params); }
+
+	Future<std::pair<Optional<StorageMetrics>, int>> waitStorageMetrics(KeyRange const& keys,
```

Comment:
Why does this make sense in the transaction processor class?
To me, the name of method/actor makes it appear like I want something from storage servers. When I dive into the implementation of this, I see that really that is a two step process. Step 1 is to get a list of locations for the key range is question. For that we, make a db access. Step 2 is to go to those servers and ask about metrics.

IMHO, this should be bundled into a StorageMetricsInterface class that DD and others can use. It would be reasonable for that in turn to have a txnProcessor that looks up the db for the key to server maps.

Reply:
This can include things like that waitMetrics, splitMetrics, readHotRanges etc

Reply:
My first thought is because it uses a real `Database cx` object though not involve transaction usage, and the DDTxnProcessor is like a proxy to `Database cx` so I include those methods in `IDDTxnProcessor`.

The reason you mention putting step 1 in txnProcessor makes sense to me. What I'm concerned about is it requires more refactoring in related methods in NativeAPI.

Reply:
After discussion, the future direction is to separate the dependency on Storage Servers and DatabaseContext. This is progressive work involving NativeAPI change.